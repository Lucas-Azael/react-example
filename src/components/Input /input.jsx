import './styles.css'

export const Input = ({searchValue, handleChange}) => {
    return (
        <div>
            <input 
            onChange={handleChange}
            value={searchValue}
            type="search" 
            placeholder="Buscar posts"
            />
        </div>
    )
}