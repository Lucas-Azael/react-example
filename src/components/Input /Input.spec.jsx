import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event";
import { Input } from "./input"

describe('<Input />', () => {
    it('should render component', () => {
        render(<Input />)
    });

    it('should have a value of searchValue', () => {
        const fn = jest.fn()
        render(<Input handleChange={fn} searchValue={'Test'}/>)
        
        const input = screen.getByPlaceholderText(/Buscar posts/i)

        expect(input.value).toBe('Test');
    });

    it('should call handleChange function on each key pressed', () => {
        const fn = jest.fn()
        render(<Input handleChange={fn} />)

        const input = screen.getByPlaceholderText(/Buscar posts/i)

        const value = 'Test do input';

        userEvent.type(input, value);
        
        expect(fn).toHaveBeenCalledTimes(value.length);
    });

    it('input value must be equal to the entered value', () => {
        const fn = jest.fn()
        render(<Input handleChange={fn} />)

        const input = screen.getByPlaceholderText(/Buscar posts/i)

        const value = 'Test do input';

        userEvent.type(input, value);
        
        expect(input.value).toBe(value);
    });

    it('should match snapshot', () => {
        const fn = jest.fn()
        const { container } = render(<Input handleChange={fn}/>);
        //O primeiro filho do container é o elemento que ele está renderizando
        expect(container.firstChild).toMatchSnapshot();
    });
})