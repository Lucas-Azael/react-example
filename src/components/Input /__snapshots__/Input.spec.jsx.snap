// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<div>
  <input
    placeholder="Buscar posts"
    type="search"
    value=""
  />
</div>
`;
